File: car-pooling.py
class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        kms = [0]*1001
        for p, f, t in trips:
            kms[f] += p
            kms[t] -= p
        ps = 0
        for km in kms:
            ps += km
            if ps > capacity:
                return False
        return True

File: car-pooling.cpp
class Solution {
public:
    bool carPooling(vector<vector<int>>& trips, int capacity) {
        vector <int> path(1001);
        for(auto &trip: trips){
            path[trip[1]] += trip[0];
            path[trip[2]] -= trip[0];
        }
        int sum_ = 0;
        for(auto n: path){
            sum_ += n;
            if (sum_ > capacity){
                return false;
            }
        }
        return true;
    }
};